lib/shopify_app/access_scopes/shop_strategy.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module ShopifyApp
+  module AccessScopes
+    class ShopStrategy
+      class << self
+        def update_access_scopes?(shop_domain)
+          shop_access_scopes = shop_access_tokens(shop_domain)
+          return true unless shop_access_scopes
```

Comment:
Does this result in infinite loops if the app developer doesn't store access scopes? Is this backwards compatible?

Reply:
No because if developer does not store access scopes, we would get a `NotImplemented` error. If this happens, the `ScopesVerification` fails (this is fine because it's opt-in). `CallbackController` and `OmniAuthConfiguration` will mark scope updates as not required and won't perform additional OAuth flows.